File: app/src/main/java/com/wearworeworn/model/Product.kt
package com.wearworeworn.model

import com.google.gson.annotations.SerializedName

data class Product(
    @SerializedName("id_product") val id: Int,
    @SerializedName("nama_product") val name: String,
    @SerializedName("deskripsi") val description: String?,
    @SerializedName("material_pakaian") val material: String?,
    @SerializedName("url_size_chart") val sizeChartUrl: String?,
    @SerializedName("harga") val price: Double,
    @SerializedName("images") val images: List<ProductImage> = emptyList(),
    @SerializedName("variants") val variants: List<ProductVariant> = emptyList(),
    @SerializedName("categories") val categories: List<Category> = emptyList()
)

data class ProductImage(
    @SerializedName("id_image") val id: Int,
    @SerializedName("id_product") val productId: Int,
    @SerializedName("url_gambar") val imageUrl: String,
    @SerializedName("is_primary") val isPrimary: Boolean
)

data class ProductVariant(
    @SerializedName("id_variant") val id: Int,
    @SerializedName("stok") val stock: Int,
    @SerializedName("size") val size: SizeInfo? = null
)

data class SizeInfo(
    @SerializedName("id_size") val id: Int,
    @SerializedName("nama_size") val name: String
)

File: database/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Product extends Model
{
    use HasFactory;

    protected $primaryKey = 'id_product';

    protected $fillable = [
        'nama_product',
        'deskripsi',
        'material_pakaian',
        'url_size_chart',
        'harga',
    ];

    protected $casts = [
        'harga' => 'decimal:2',
    ];

    public function categories(): BelongsToMany
    {
        return $this->belongsToMany(
            Category::class,
            'product_category',
            'id_product',
            'id_category'
        );
    }

    public function variants(): HasMany
    {
        return $this->hasMany(ProductVariant::class, 'id_product', 'id_product');
    }

    public function images(): HasMany
    {
        return $this->hasMany(ProductImage::class, 'id_product', 'id_product');
    }

    public function primaryImage(): HasOne
    {
        return $this->hasOne(ProductImage::class, 'id_product', 'id_product')
                    ->where('is_primary', 1);
    }

    public function getVariantBySize(string $size): ?ProductVariant
    {
        return $this->variants()
                    ->whereHas('size', fn($q) => $q->where('nama_size', $size))
                    ->first();
    }
}
